service/history/queueFactoryBase.go

```diff
@@ -127,16 +130,22 @@ func QueueSchedulerRateLimiterProvider(
 func QueueFactoryLifetimeHooks(
 	params QueueFactoriesLifetimeHookParams,
 ) {
+	var factories []QueueFactory
+	for _, f := range params.Factories {
+		if f.Enabled() {
+			factories = append(factories, f)
+		}
+	}
 	params.Lifecycle.Append(
```

Comment:
- This lifecycle is only for starting & stopping the host level worker pool for each queue, not the actual queue processing logic for loading tasks. I thought we need a configuration to stop the queue itself not just the worker pool?
- Currently worker pool can already be turned off dynamically (without restart), by setting the worker count to 0.